homeassistant/components/watchdog_file_watcher.py

```diff
@@ -0,0 +1,75 @@
+"""
+Component for monitoring activity on a folder.
+"""
+import asyncio
+import logging
+import voluptuous as vol
+from watchdog.observers import Observer
+from watchdog.events import PatternMatchingEventHandler
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+#DEPENDENCIES = ['watchdog']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PATH = 'folder'
+DOMAIN = "watchdog_file_watcher"
+EVENT_TYPE = "event_type"
+SRC_PATH = "src_path"
+PATTERNS = ["*.txt", "*.py", "*.md", "*.jpg", "*.png"]
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_PATH): cv.isdir})
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the folder watcher."""
+    conf = config.get(DOMAIN)
+    path = conf.get(CONF_PATH)
+    if not hass.config.is_allowed_path(path):
+        _LOGGER.error("folder %s is not valid or allowed", path)
+    else:
+        Watcher(path, hass)
```

Comment:
I'm not entirely sure how entities are added for components, but when does this entity get added to hass?

Also, why is this a component and not a `sensor` platform?

Reply:
Im not sure about this too, but it appears to work..